heron/scheduler-core/src/java/org/apache/heron/scheduler/RuntimeManagerRunner.java

```diff
@@ -365,6 +405,39 @@ protected void cleanState(
     LOG.fine("Cleaned up topology state");
   }
 
+  @VisibleForTesting
+  PackingPlans.PackingPlan buildNewPackingPlan(PackingPlans.PackingPlan currentProtoPlan,
```

Comment:
Instead of replicating most of the code from the `buildNewPackingPlan(cp, cr, cn, t)`, can we reuse that method? Providing a special number for `cn` say -1, to indicate the old `repack` needs to be used.